frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx

```diff
@@ -0,0 +1,93 @@
+import { CheckIcon, CopyIcon } from "@radix-ui/react-icons";
+import { useEffect, useRef, useState } from "react";
+
+import { Button } from "@/components/ui/button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { RequestorState } from "../../reducer";
+import { getBodyValue } from "./utils";
+
+export type CopyAsCurlProps = Pick<
+  RequestorState,
+  "body" | "method" | "path" | "requestHeaders" | "queryParams"
+>;
+
+/**
+ * Copy the current request as a cURL command to the clipboard.
+ */
+export function CopyAsCurl({
+  body,
+  method,
+  path,
+  queryParams,
+  requestHeaders,
+}: CopyAsCurlProps) {
+  const [isCopied, setIsCopied] = useState(false);
+  const timeout = useRef<ReturnType<typeof setTimeout>>();
+  const isUnsupportedBodyType =
+    body.type === "file" || (body.type === "form-data" && body.isMultipart);
+
+  const handleCopy = async () => {
+    const payload = getBodyValue({ body, method });
+
+    const headers = requestHeaders.reduce((acc, { enabled, key, value }) => {
+      if (!enabled) {
+        return acc;
+      }
+
+      return `${acc} -H "${key}: ${value}"`;
+    }, "");
+
+    const url = new URL(path);
+    for (const { enabled, key, value } of queryParams) {
+      if (enabled) {
+        url.searchParams.append(key, value);
+      }
+    }
+
+    const data = payload ? `-d ${payload}` : "";
+    const curlCommand = `curl -X ${method} ${url} ${headers} ${data}`;
```

Comment:
might wanna put the url in quotes, especially if there's a querystring attached, since most shells need a `?` to be wrapped in quotes!

to test locally, try making a GET request against a route with some query params attached

Reply:
Good one! Will add